app/domain/authentication/authn_gcp/update_authenticator_input.rb

```diff
@@ -0,0 +1,105 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGcp
+
+    UpdateAuthenticatorInput = CommandClass.new(
+      dependencies: {
+        verify_and_decode_token: Authentication::OAuth::VerifyAndDecodeToken.new,
+        decoded_token_class:     DecodedToken,
+        logger:                  Rails.logger
+      },
+      inputs:       [:authenticator_input]
+    ) do
+
+      extend Forwardable
+      def_delegators :@authenticator_input, :authenticator_name, :account, :credentials, :username
+
+      def call
+        validate_token
+        validate_conjur_username
+        updated_input
+      end
+
+      private
+
+      def validate_token
+        decoded_token
+        validate_audience
+      end
+
+      def decoded_token
+        @decoded_token ||= @decoded_token_class.new(
+          decoded_token_hash: @verify_and_decode_token.call(
+            provider_uri:     PROVIDER_URI,
+            token_jwt:        decoded_credentials.jwt,
+            claims_to_verify: {
+              verify_iss:        true,
+              iss:               ISS,
+              verify_iat:        true,
+              verify_expiration: true
+            }
+          ),
+          logger:             @logger
+        )
+      end
+
+      def decoded_credentials
+        @decoded_credentials ||= Authentication::Jwt::DecodedCredentials.new(credentials)
+      end
+
+      def validate_audience
+        if audience_parts.length != 3 ||
+          audience_parts[0] != "conjur" ||
+          audience_parts[1] != account
+          raise Errors::Authentication::AuthnGcp::InvalidAudience, audience
+        end
+      end
+
+      def audience_parts
+        @audience_parts ||= audience.split('/')
+      end
+
+      def audience
+        @audience ||= @decoded_token.audience
+      end
+
+      def validate_conjur_username
+        if conjur_username == "admin"
+          raise Errors::Authentication::AdminAuthenticationDenied, authenticator_name
+        end
+
+        @logger.debug(
+          LogMessages::Authentication::Jwt::ExtractedUsernameFromToken.new(
+            conjur_username
+          )
+        )
+      end
+
+      def conjur_username
+        @conjur_username ||= audience_parts[2]
```

Comment:
We should add here an input validation to prevent code injection to the logs,
For example if aud = conjur/account/<inject malicious content>

Reply:
what will we check in this case?

Reply:
Maybe the length and white list characters, @sashaCher WDYT?